File: admin/src/api/payment.ts
import { http, type ApiEnvelope } from '../utils/request'
import type { PaginatedData } from './admin-user'
import type { OrderUserSummary } from './order'

export type PaymentProvider = 'alipay' | 'wechat'
export type PaymentMethod = 'alipay_page' | 'alipay_wap' | 'wechat_native' | 'wechat_h5'
export type PaymentStatus = 'pending' | 'paid' | 'closed' | 'failed' | 'refunded'
export type RefundStatus = 'pending' | 'succeeded' | 'failed'

export interface AdminPaymentItem {
  payment_no: string
  order_no: string
  user: OrderUserSummary
  provider: PaymentProvider
  method: PaymentMethod
  status: PaymentStatus
  amount_cents: number
  currency: string
  expires_at: string
  paid_at: string | null
  created_at: string
  order_status: string
  order_type: 'purchase' | 'renewal'
}

export interface AdminPaymentDetail extends AdminPaymentItem {
  upstream_trade_no: string | null
  last_error_message: string | null
  refund: AdminRefundItem | null
}

export interface AdminRefundItem {
  refund_no: string
  payment_no: string
  order_no: string
  user: OrderUserSummary
  provider: PaymentProvider
  status: RefundStatus
  amount_cents: number
  currency: string
  reason: string
  created_at: string
  completed_at: string | null
  failed_at: string | null
}

export async function getPayments(params?: Record<string, unknown>) {
  const response = await http.get<ApiEnvelope<PaginatedData<AdminPaymentItem>>>('/payments', { params })
  return response.data.data
}

export async function getPaymentDetail(paymentNo: string) {
  const response = await http.get<ApiEnvelope<AdminPaymentDetail>>(`/payments/${paymentNo}`)
  return response.data.data
}

export async function syncPayment(paymentNo: string) {
  const response = await http.post<ApiEnvelope<AdminPaymentDetail>>(`/payments/${paymentNo}/sync`)
  return response.data.data
}

export async function createPaymentRefund(paymentNo: string, reason: string) {
  const response = await http.post<ApiEnvelope<AdminRefundItem>>(`/payments/${paymentNo}/refunds`, { reason })
  return response.data.data
}

export async function retryPaymentProvision(paymentNo: string) {
  const response = await http.post<ApiEnvelope<AdminPaymentDetail>>(`/payments/${paymentNo}/retry-provision`)
  return response.data.data
}

export async function getRefunds(params?: Record<string, unknown>) {
  const response = await http.get<ApiEnvelope<PaginatedData<AdminRefundItem>>>('/refunds', { params })
  return response.data.data
}

File: server/internal/usecase/admin/dto/payment.go
package dto

import "time"

type PaymentListQuery struct {
	Page        int    `form:"page" validate:"omitempty,min=1"`
	PerPage     int    `form:"per_page" validate:"omitempty,min=1,max=100"`
	Provider    string `form:"provider" validate:"omitempty,oneof=alipay wechat"`
	Method      string `form:"method" validate:"omitempty,oneof=alipay_page alipay_wap wechat_native wechat_h5"`
	Status      string `form:"status" validate:"omitempty,oneof=pending paid closed failed refunded"`
	OrderNo     string `form:"order_no" validate:"omitempty,max=64"`
	PaymentNo   string `form:"payment_no" validate:"omitempty,max=64"`
	UserKeyword string `form:"user_keyword" validate:"omitempty,max=128"`
	DateFrom    string `form:"date_from" validate:"omitempty,max=32"`
	DateTo      string `form:"date_to" validate:"omitempty,max=32"`
}

type RefundListQuery struct {
	Page      int    `form:"page" validate:"omitempty,min=1"`
	PerPage   int    `form:"per_page" validate:"omitempty,min=1,max=100"`
	Provider  string `form:"provider" validate:"omitempty,oneof=alipay wechat"`
	Status    string `form:"status" validate:"omitempty,oneof=pending succeeded failed"`
	OrderNo   string `form:"order_no" validate:"omitempty,max=64"`
	PaymentNo string `form:"payment_no" validate:"omitempty,max=64"`
	RefundNo  string `form:"refund_no" validate:"omitempty,max=64"`
	DateFrom  string `form:"date_from" validate:"omitempty,max=32"`
	DateTo    string `form:"date_to" validate:"omitempty,max=32"`
}

type PaymentItem struct {
	PaymentNo   string           `json:"payment_no"`
	OrderNo     string           `json:"order_no"`
	User        OrderUserSummary `json:"user"`
	Provider    string           `json:"provider"`
	Method      string           `json:"method"`
	Status      string           `json:"status"`
	AmountCents uint64           `json:"amount_cents"`
	Currency    string           `json:"currency"`
	ExpiresAt   time.Time        `json:"expires_at"`
	PaidAt      *time.Time       `json:"paid_at"`
	CreatedAt   time.Time        `json:"created_at"`
	OrderStatus string           `json:"order_status"`
	OrderType   string           `json:"order_type"`
}

type PaymentDetail struct {
	PaymentItem
	UpstreamTradeNo  *string     `json:"upstream_trade_no"`
	LastErrorMessage *string     `json:"last_error_message"`
	Refund           *RefundItem `json:"refund"`
}

type RefundItem struct {
	RefundNo    string           `json:"refund_no"`
	PaymentNo   string           `json:"payment_no"`
	OrderNo     string           `json:"order_no"`
	User        OrderUserSummary `json:"user"`
	Provider    string           `json:"provider"`
	Status      string           `json:"status"`
	AmountCents uint64           `json:"amount_cents"`
	Currency    string           `json:"currency"`
	Reason      string           `json:"reason"`
	CreatedAt   time.Time        `json:"created_at"`
	CompletedAt *time.Time       `json:"completed_at"`
	FailedAt    *time.Time       `json:"failed_at"`
}

type RefundCreateRequest struct {
	Reason string `json:"reason" validate:"required,max=500"`
}
